th2_data_services_lwdp/commands/http.py

```diff
@@ -120,7 +120,7 @@ def _data_object(self, data_source: HTTPDataSource) -> Data:
              Data
         """
         sse_events_stream = partial(self._sse_events_stream, data_source)
-        source = partial(self._sse_handler.handle_stream, sse_events_stream)
+        source = partial(self._sse_handler.handle, sse_events_stream)
 
         return Data(source, cache=self._cache)
```

Comment:
The idea to chenge to 
`
sse_events_stream = partial(self._sse_events_stream, data_source)
return Data(sse_events_stream).map_stream(self._sse_handler).use_cache(self._cache)
`

Reply:
Didnt we change method to return partial object because Data has to have correct data type?

Reply:
`Didnt we change method to return partial object because Data has to have correct data type?`

We have to use partials to put `generator function`  to Data object.
My idea is to use `map_stream` with adapter (as we implemented some days ago) instead of using `partial`.

`If I remember correctly we returned such object but we used apply_adapter instead of map_stream. Wouldnt we be doing the same thing?`

we used `apply_adapter ` to apply users adapters that user could apply to commands
e.g. 
command.apply_adapter(Adapter().handler)

We changed this behavior. Now the user will apply adapter to Data object.
data.map(adapter())

Reply:
Updated